pgx/src/spi.rs

```diff
@@ -37,25 +37,61 @@ pub enum SpiOk {
     TdRegister = 17,
 }
 
-#[derive(Debug, Primitive)]
+#[derive(Debug, PartialEq)]
+#[repr(i32)]
 pub enum SpiError {
-    // NB:  These are #define'd as negative, but we redefine them as positive so that
-    // #[derive(Primitive)] will work.  We just need to negate result codes from the
-    // various SPI_xxx functions when looking for errors
-    Connect = 1,
-    Copy = 2,
-    Opunknown = 3,
-    Unconnected = 4,
+    Null = 0,
```

Comment:
This one is mildly odd as there are functions that will return a nullptr to signal errors happened, and the result code gets set to 0 at the start of most functions that set the result code. So it's not a formal "variant", but if we see it, it suggests something has gone wrong.

Reply:
Is it advisable to even use a nullptr to signal "something bad happened"? Or should it consistently be `Ok/Err` wherever it's applicable?

Reply:
I would say "no" but this is part of what Postgres does. However it's not actually clear it ever arises in a codepath we have hit yet, so thinking about it, I don't know if we should.

cc @willmurnane